components/brave_new_tab_ui/containers/newTab/settings/solidColorChooser.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import { StyledCustomBackgroundSettings } from '../../../components/default'
+import NavigateBack from '../../../components/default/settings/navigateBack'
+import SolidColorBackgroundOption from './solidColorBackgroundOption'
+import { solidColorsForBackground } from '../../../data/colors'
+import { getLocale } from '$web-common/locale'
+interface Props {
+  currentColor?: string
+  useSolidColorBackground: (color: string) => void
+  onBack: () => void
+}
+
+class SolidColorChooser extends React.PureComponent<Props, {}> {
```

Comment:
Similarly to before, might be worth converting this to a functional component too.

```tsx
const containerEl = useRef<HTMLDivElement>();
useEffect(() => {
    containerEl.current?.scrollIntoView(true);
}, []); // Empty deps to only run on mount.

return <div ref={containerEl}>
    ...
</div>;
```